File: Code/Chapter 13/VB.NET/SqlServerUDF/GetRandomNumber.vb
Imports System
Imports System.Data
Imports System.Data.Sql
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server

Partial Public Class UserDefinedFunctions
   <Microsoft.SqlServer.Server.SqlFunction()> _
   Public Shared Function GetRandomNumber2() As SqlInt32
      Dim rnd As Random = New Random
      Return rnd.Next()
   End Function
End Class

File: Code/Chapter 13/C#/SqlServerUDF/GetRandomNumber.cs
using System;
using System.Data;
using System.Data.Sql;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

public partial class SqlServerUDF
{
   [Microsoft.SqlServer.Server.SqlFunction]
   public static SqlInt32 GetRandomNumber2()
   {
      Random rnd = new Random();
      return rnd.Next();
   }
};